monte_carlo.py

```diff
@@ -23,60 +27,229 @@
+            xmind = bounds.get_x_min() - neutron_position.x
+            ymaxd = bounds.get_y_max() - neutron_position.y
+            ymind = bounds.get_y_min() - neutron_position.y
+            zmaxd = bounds.get_z_max() - neutron_position.z
+            zmind = bounds.get_z_min() - neutron_position.z
+               
+            # lim_bound contains the string of the limiting boundary
+            lim_bound = list()
+                
+            # tempd contains the current smallest r
+            tempd = neutron_distance
+                
+            # r is a variable that contains the distance along the
+            # direction vector to the boundary being tested 
+            #test xmax
+            r = xmaxd/(sin(theta)*cos(phi))
+            if (0 <= r and r < neutron_distance and not r == -0.0):
+                tempd = r
+                lim_bound.append('x_max')
+                
+            #test xmin
+            r = xmind/(sin(theta)*cos(phi))
+            if (0 <= r and r < neutron_distance and \
+                    r < tempd and not r == -0.0):
+                tempd = r
+                lim_bound = []
+                lim_bound.append('x_min')
+            elif r == tempd:
+                lim_bound.append('x_min')
+            
+            #test ymax
+            r = ymaxd/(sin(theta)*sin(phi))
+            if (0 <= r and r < neutron_distance and \
+                    r < tempd and not r == -0.0):
+                tempd = r
+                lim_bound = []
+                lim_bound.append('y_max')
+            elif r == tempd:
+                lim_bound.append('y_max')               
+                
+            #test ymin
+            r = ymind/(sin(theta)*sin(phi))
+            if (0 <= r and r < neutron_distance and \
+                    r < tempd and not r == -0.0):
+                tempd = r
+                lim_bound = []
+                lim_bound.append('y_min')
+            elif r == tempd:
+                lim_bound.append('y_min')               
+            
+            #test zmax
+            r = zmaxd/cos(theta)
+            if (0 <= r and r < neutron_distance and \
+                    r < tempd and not r == -0.0):
+                tempd = r
+                lim_bound = []
+                lim_bound.append('z_max')
+            elif r == tempd:
+                lim_bound.append('z_max')              
+                
+            #test zmin
+            r = zmind/cos(theta)
+            if (0 <= r and r < neutron_distance and \
+                    r < tempd and not r == -0.0):
+                tempd = r
+                lim_bound = []
+                lim_bound.append('z_min')
+            elif r == tempd:
+                lim_bound.append('z_min')               
+                
+            # if the neutron stays within the bounding box
+            if lim_bound == []:
+                neutron_distance = 0
+            neutron_position.sadd(neutron_movement_unit_vector * tempd)
```

Comment:
Instead of using ``sadd`` I would write:
```
neutron_position += neutron_movement_unit_vector * tempd
```
for improved readability (it's not too clear what ``sadd`` does)